base_wamas_ubl/lib/wamas/wamas2ubl.py

```diff
@@ -121,8 +121,9 @@ def dict2ubl(msg_type, data, extra_data=False):
         extractor.get_line("KRETPQ", "IvKretp_KretId_KretNr")
     elif msg_type == "PickingResponse":
         extractor.get_head("AUSKQ", "IvAusk_AusId_AusNr")
-        extractor.get_head("WATEKQ", "IvTek_TeId", extractor.packages)
-        extractor.get_line("WATEPQ", "IvAusp_UrAusId_AusNr", "IvTep_TeId")
+        if "WATEKQ" in extractor.data and "WATEPQ" in extractor.data:
+            extractor.get_head("WATEKQ", "IvTek_TeId", extractor.packages)
+            extractor.get_line("WATEPQ", "IvAusp_UrAusId_AusNr", "IvTep_TeId")
```

Comment:
what if we have AUSPQ lines for different pickings?

Reply:
I think a better strategy would be to retrieve lines for AUSPQ as well, but mark the DespatchAdvice lines as cancellations somehow

Reply:
For now, [base_wamas_ubl](https://github.com/OCA/edi/tree/16.0/base_wamas_ubl/lib/wamas/wamas_grammar) didn't define `auspq` grammar, would you want to create new one?

>what if we have AUSPQ lines for different pickings?

In `allo_despatch_advice_import` is default that all lines reference to 1 specific picking : [here](https://github.com/camptocamp/alloboissons_odoo/blob/e534f43676cc9ca9afe17aa56d7ef46daeabcfd8/odoo/local-src/allo_despatch_advice_import/wizard/despatch_advice_import.py#L132) (only get the name from the `order`)

In OCA: it will compare: [here](https://github.com/QuocDuong1306/edi/blob/bea9b74a9b70a211e53c8780fa3502422d634a87/despatch_advice_import/wizard/despatch_advice_import.py#L182)

Reply:
one DespatchAdvice = one picking yes
but one WAMAS can contain multiple pickings that we split in different DespatchAdvice

so we might have one WAMAS with multiple AUSPQ lines for different pickings -> multiple DespatchAdvice